### PR TITLE
Mediawiki tests

### PR DESCRIPTION
**Pull request by clarakosi**
Bug: [T225614](https://phabricator.wikimedia.org/T225614)

### CODE REVIEW

File: examples/Mediawiki/PresaveTransform.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
+suite: Pre-save Transform
+description: Testing preSaveTransform() function
+type: RPC
+
+tests:
+  - description: Create page with pipetrick and check if transformed
```
Comment (brightbyte) [inline examples/Mediawiki/PresaveTransform.yaml:6]: Oooohhhh, pipetrick - advanced wikitext magic ;)

File: examples/Mediawiki/MoveAPI.yaml [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+          form-data:
+            reason: misspelling
+            token: +\
+        response:
+          body:
+            move:
+              from: MoveWithoutt
+              to: Move without
+              reason: misspelling
+              redirectcreated: ""
```
Comment (brightbyte) [inline examples/Mediawiki/MoveAPI.yaml]: Needs a check that no redirect was created and that the talk page and the subpage were not moved. You can do with a single request, action=query&titles=MoveWithoutt|Talk:MoveWithoutt|MoveWithoutt/subpage, but you need to hard code the page IDs for now. For MoveWithoutt (which should not be found) the key "-1" should be used. Non-existing pages are indexed as -1, -2, -3 etc in the response.

File: examples/Mediawiki/MoveAPI.yaml [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+      # move page with redirect, subpage, and talkpage
+      - request:
+          method: post
+          path: api.php
+          parameters:
+            action: move
+            from: MoveWithh
+            to: Move With
+            movetalk: ""
+            movesubpages: ""
```
Comment (brightbyte) [inline examples/Mediawiki/MoveAPI.yaml:165]: requires the move-subpages permission

File: examples/Mediawiki/MoveAPI.yaml [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+          response:
+            body:
+              edit:
+                result: Success
+      # move page without redirect, subpage, and talkpage
+      - request:
+          method: post
+          path: api.php
+          parameters:
+            action: move
```
Comment (brightbyte) [inline examples/Mediawiki/MoveAPI.yaml:72]: requires the move permission

File: examples/Mediawiki/MoveAPI.yaml [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+        response:
+          body:
+            move:
+              from: MoveWithh
+              to: Move With
+              reason: misspelling
+              talkfrom: Talk:MoveWithh
+              talkto: Talk:Move With
+              subpagefrom: MoveWithh
+              subpageto: Move With
```
Comment (brightbyte) [inline examples/Mediawiki/MoveAPI.yaml]: is that really the response format? i didn't check, but... how would that work if multiple subpages get moved?

File: examples/Mediawiki/QueryAPI.yaml [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+  - description: Fetch links of a page
+    interaction:
+      - request:
+          path: api.php
+          parameters:
+            action: query
+            prop: links|categories|templates
+            titles: Query API
+            format: json
+        response:
```
Comment (brightbyte) [inline examples/Mediawiki/QueryAPI.yaml:34]: Haven't tried, but I think yaml supports the entire path in a single line, as in:
    response: body: query: pages:

Might be more readyble

File: examples/Mediawiki/MoveAPI.yaml (hunk 3 above)
Comment (clarakosi) [reply examples/Mediawiki/MoveAPI.yaml:165]: ahh ok!

File: examples/Mediawiki/MoveAPI.yaml (hunk 5 above)
Comment (clarakosi) [reply examples/Mediawiki/MoveAPI.yaml]: No, it's mostly a place holder.

File: examples/Mediawiki/MoveAPI.yaml (hunk 4 above)
Comment (clarakosi) [reply examples/Mediawiki/MoveAPI.yaml:72]: Yup. Mentioned it at the top of the file.

File: examples/Mediawiki/QueryAPI.yaml (hunk 6 above)
Comment (clarakosi) [reply examples/Mediawiki/QueryAPI.yaml:34]: Will try

Comment (clarakosi) [reply examples/Mediawiki/QueryAPI.yaml:34]: Tried it with one of the responses and symfony's yaml parser throws an error ` A colon cannot be used in an unquoted mapping value at line 269 (  
  near "response: body: edit: result: Success").`

Comment (brightbyte) [reply examples/Mediawiki/QueryAPI.yaml:34]: Meh. This would have been so much nicer ;)

Comment (clarakosi) [reply examples/Mediawiki/QueryAPI.yaml:34]: I agree

**Pull request merged**